### ISSUE TITLE
Run DANNCE on Rat7M 

### ISSUE THREAD

**Issue by EvenGu** [spoonsso/dannce#109]
Hi,

I want to run DANNCE model on the Rat7M dataset as a baseline. To do that, I am preparing the *_dannce.mat file. Rat7M comes with the Camera parameters and the 3D groundtruth. But I don't know how to fill the field of `labelData.data_2d` for each of the six cameras.
Could you tell me how to get those values? ---- I thought DANNC does NOT need 2d labels but without these values the model ran into errors...
BTW, I am going to use the pretrained models (DANNCE and COM) you provided in the `demo/markless_mouse_1/`, please let me know if you suggest otherwise.

Thank you in advance.

**Comment by spoonsso** [comment]
The way it is currently written, all of our models (including ones that do not require 2D labels, such as DANNCE) share a common set of data loading functions and thus expect to find `labelData.data_2d`. It is not ideal behavior, and I hope that a contributor at some point improves this =) If you are just using a DANNCE model, you can just fill data_2d with zeroes.

Are you planning to train on Rat7M or just do prediction on Rat7M? For prediction, yes, the pretrained weights in `demo/markerless_mouse_1/DANNCE/weights/weights.rat.AVG.6cam.hdf5` should be fine. Remember to set `vol_size: 240` to use 3D volumes that are large enough to contain a rat and set `new_n_channels_out: 22` (or scrub `new_n_channels_out` from the config entirely).

**Comment by EvenGu** [comment]
Thank you for your fast reply!
I changed the `new_n_channels_out` to 20 since there are 20 keypoints in the Rat7M MoCap data. I will change the `vol_size` to 240. 
I can try to modify the data loading process for DANNCE. I can make a pull request if I get success.
Happy Danncing!

**Comment by spoonsso** [comment]
Thanks! We are preparing a new release soon, if you want to make any
changes I recommend branching off release_development.

But also -- you are just planning to do Rat7M prediction, right? If you are
planning to try training, there is more to discuss re: logistics (e.g., you can't fit all of the training data into memory so need to pregenerate all of the volumes and save them to disk -- but there is a config option for this).

**Comment by EvenGu** [comment]
Hi, Dr. Dunn,
Yes that I plan to just do the Rat7M predictions. I first run the COM prediction using the `weights.rat.COM.hdf5` in the demo/markerless_mouse_1/COM/weights/. I used the `debug` mode and output the COM overlays. Unfortunately, the overlay looks quite off ( I randomly picked Camera5.)  I realized that I cannot use the flag `--com-fromlabels` to do DANNCE predictions so I am training the COM from scratch, which gives me the error about memory loading, just as you expected.
May I ask how to pregenerate all of the volumes and save them to disk?

**Comment by spoonsso** [comment]
What downfac did you use for the COM prediction? It is quite sensitive to
using the same value that was used for training.

**Comment by EvenGu** [comment]
I used `downfac: 4`.  In the provided config files, the values are 
in `com_mouse_config.yaml ` downfac=**2** and 
in `dannce_rig_com_config.yaml` downfac=**8** 
What number should I use? **8**? Can I set the downfac to 8 and then predict COM using the provided `weights.rat.COM.hdf5`  for Rat7M? Or do I need to train the COM first?

**Comment by spoonsso** [comment]
Try it with downfac 2

**Comment by spoonsso** [comment]
Prediction, that is. With the rat.com weights
